### PR TITLE
Import homebrew/dupes into core.

### PR DESCRIPTION
**Pull request by MikeMcQuaid**
Part of the tap consolidation process. Related Homebrew/brew PR for checking that some of
these aren't depended on in https://github.com/Homebrew/brew/pull/2365.

### CODE REVIEW

**Comment by MikeMcQuaid** [comment]
Let's hold off merging this until we have a new tag with https://github.com/Homebrew/brew/pull/1732.

**Comment by tdsmith** [comment]
I merged an openssh upgrade into homebrew-dupes this morning & @JCount pointed out that you might not have been expecting that; sorry! https://github.com/Homebrew/homebrew-dupes/commit/628b3cd8833d74c28acf79a596ab971609dbf33a

**Comment by MikeMcQuaid** [comment]
@tdsmith A delayed: no worries!

**Comment by fxcoudert** [comment]
```
==> Verifying tcl-tk--tcllib-1.18.tar.gz checksum
Error: SHA256 mismatch
Expected: 6a87881f545afb69c1130f60984b5d35cc22f1593b0835b982871c188fde3de8
Actual: 9988b4385403c2aac78743fd3fce2d22e82686a56e6ca25942cb83c7d9e641db
```

What is even weirder is that the original formula in `dupes` had yet another shasum:

```
    url "https://downloads.sourceforge.net/project/tcllib/tcllib/1.18/tcllib-1.18.tar.gz"
    sha256 "72667ecbbd41af740157ee346db77734d1245b41dffc13ac80ca678dd3ccb515"
```

File: Formula/srm.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  sha256 "7583c1120e911e292f22b4a1d949b32c23518038afd966d527dae87c61565283"
+
+  bottle do
+    cellar :any_skip_relocation
+    sha256 "3965dac338dab92d3c25306ec55dda0b1b75b78fb1f9060cd2e1c7b1e842f7a5" => :el_capitan
+    sha256 "bad23b6a0f2eebf37cc5b9f02f0001f5bf5af46fab55dd67aeb7b67796891b90" => :yosemite
+    sha256 "800ceccc9adecec1ceabfbc6c3d2aaa1da682db174ab02ae75cbcbba4cee1270" => :mavericks
+  end
+
+  keg_only :provided_by_osx
```
Comment (fxcoudert) [inline Formula/srm.rb]: `if MacOS.version < :sierra` since this is was removed in Sierra

File: Formula/heimdal.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+class Heimdal < Formula
+  desc "Free Kerberos 5 implementation"
+  homepage "https://www.h5l.org"
+  url "https://www.h5l.org/dist/src/heimdal-1.6rc2.tar.gz"
```
Comment (fxcoudert) [inline Formula/heimdal.rb]: There is now a 7.1.0 release

File: Formula/bc.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
+class Bc < Formula
+  desc "arbitrary precision numeric processing language"
+  homepage "https://www.gnu.org/software/bc/"
+  url "https://ftpmirror.gnu.org/bc/bc-1.06.tar.gz"
```
Comment (fxcoudert) [inline Formula/bc.rb]: There is now a 1.07

**Comment by chdiza** [comment]
I'm wondering whether `bzip` should be dropped.  It hasn't been updated since 2010 and I'm pretty sure that every Mac from Snow Leopard (or perhaps Lion at the outside) and up has the latest version of `bzip`.

File: Formula/srm.rb (hunk 1 above)
Comment (ilovezfs) [inline Formula/srm.rb]: I think we should probably not import this at all since apple removed it to prevent people from killing SSDs.

Comment (chdiza) [inline Formula/srm.rb]: Yeah, but lots of people still have spinning platters, either in their machines or as external disks.

Comment (ilovezfs) [inline Formula/srm.rb]: say it isn't so

File: Formula/python.rb [hunk 4]
```diff
@@ -32,7 +32,7 @@ class Python < Formula
   depends_on "sqlite" => :recommended
   depends_on "gdbm" => :recommended
   depends_on "openssl"
-  depends_on "homebrew/dupes/tcl-tk" => :optional
+  depends_on "tcl-tk" => :optional
   depends_on "berkeley-db@4" => :optional
   depends_on :x11 if build.with?("tcl-tk") && Tab.for_name("homebrew/dupes/tcl-tk").with?("x11")
```
Comment (ilovezfs) [inline Formula/python.rb]: needs the homebrew/dupes reference removed here too

**Comment by MikeMcQuaid** [comment]
> nano: build should be attempted without the ncurses dep

Fails, it wants unicode supporting ncurses (which seems a good enough reason).

**Comment by MikeMcQuaid** [comment]
> tcpdump: build should be attempted without the libpcap dep

I think given this is optional it's OK?

**Comment by MikeMcQuaid** [comment]
> awk: needs ENV.deparallelize

Test needs updated, no? No idea how to do that; doesn't work like my normal `awk` does...

**Comment by ilovezfs** [comment]
analytics for last 30 days on tcpdump

* homebrew/dupes/tcpdump
329 installs
* homebrew/dupes/tcpdump --with-libpcap
4 installs
* homebrew/dupes/tcpdump --HEAD --with-libpcap
2 installs

So
>I think given this is optional it's OK?

yes. or drop the option.

**Comment by ilovezfs** [comment]
>Test needs updated, no? No idea how to do that; doesn't work like my normal awk does...

Nope the test is fine. The build must be deparallelized or the result is a horribly broken end binary. (generated files + makefile that assumes -j1 = :boom:)

The entire build takes a matter of seconds so it's fine to deparallelize IMO.

**Comment by fxcoudert** [comment]
Reproduced and reported the LAPACK issue upstream: https://github.com/Reference-LAPACK/lapack/issues/139

**Comment by ilovezfs** [comment]
CC @dpo regarding LAPACK issue.

**Comment by ilovezfs** [comment]
>Fails, it wants unicode supporting ncurses (which seems a good enough reason).

Yup.

**Comment by MikeMcQuaid** [comment]
> I'm trying a clean build of lapack 3.7.0 with gfortran 6.3.0 and Xcode 8.3, with cmake .. -DLAPACKE:BOOL=ON. It fails with:

@fxcoudert @ilovezfs should we just not pass that? Is it still useful without?

**Comment by fxcoudert** [comment]
It's the C interface, so it's typically *very* useful :)

**Comment by MikeMcQuaid** [comment]
> It's the C interface, so it's typically very useful :)

I don't C where you're going with this 🥁 :trollface:

File: Formula/tcl-tk.rb [hunk 5]
```diff
@@ -62,13 +58,7 @@ def install
         args = ["--prefix=#{prefix}", "--mandir=#{man}", "--with-tcl=#{lib}"]
         args << "--enable-threads" if build.with? "threads"
         args << "--enable-64bit" if MacOS.prefer_64_bit?
-
-        if build.with? "x11"
-          args << "--with-x"
-        else
-          args << "--enable-aqua=yes"
```
Comment (fxcoudert) [inline Formula/tcl-tk.rb]: Why did you remove `--enable-aqua=yes`?

Comment (MikeMcQuaid) [inline Formula/tcl-tk.rb]: I think that'll fix the audit failure, we'll see.

File: Formula/angband.rb [hunk 6]
```diff
@@ -16,7 +16,7 @@ class Angband < Formula
   depends_on "autoconf" => :build
   depends_on "automake" => :build
   depends_on :x11 => :optional
-  depends_on "homebrew/dupes/tcl-tk" => "with-x11" if build.with? :x11
+  depends_on "tcl-tk" => "with-x11" if build.with? :x11
```
Comment (fxcoudert) [inline Formula/angband.rb]: We should remove that entirely, since we don't offer `tcl-tk` with X11 anymore…

Comment (MikeMcQuaid) [inline Formula/angband.rb]: Done, thanks.

File: Formula/heimdal.rb (hunk 2 above)
Comment (MikeMcQuaid) [inline Formula/heimdal.rb]: Going to punt on it for now just to get this merged but: thanks.

File: Formula/tcl-tk.rb (hunk 5 above)
Comment (MikeMcQuaid) [inline Formula/tcl-tk.rb]: It broke install instead 😁. Re-added.

**Comment by MikeMcQuaid** [comment]
> It's the C interface, so it's typically very useful :)

Going to punt on `lapack` for now to get this merged before upstream figures something out. There's an argument that if it's been broken on Sierra for a while with no reports then that's a candidate for deletion.

**Comment by fxcoudert** [comment]
What is weird is that it does have a Sierra bottle, so it worked at some point. I actually wonder if it's not a bug in `cmake 3.7.2`.

**Comment by MikeMcQuaid** [comment]
> What is weird is that it does have a Sierra bottle, so it worked at some point. I actually wonder if it's not a bug in cmake 3.7.2.

Yeh, could be.

**Comment by fxcoudert** [comment]
Lapack compiles (with same options) with cmake-3.6.3.

**Comment by fxcoudert** [comment]
lapack is taken care of in #12033

**Comment by MikeMcQuaid** [comment]
Pulled without `angband` changes and `heimdal` inclusion.

**Comment by DomT4** [comment]
Probably a known thing but after something like this PR running `brew bundle cleanup` wants to uninstall all your dupe-installed formulae.

Not sure there's an easy fix given if you've installed something from a tap that duplicates a core formula you probably want to keep using the tap formula rather than matching the shorter core name & having that installed, but, raising it as an FYI.

**Comment by MikeMcQuaid** [comment]
> Not sure there's an easy fix

It may be doable in Homebrew/bundle but it'd probably be low priority until someone gets confused enough by it. Eventually you'll want to rewrite your Brewfile anyway so chances are this will motivate you to do it.
